src/components/Movie/Movie.js

```diff
@@ -1,47 +1,35 @@
 import React, { Component } from "react";
 import { withRouter } from 'react-router-dom'
+import { connect } from "react-redux";
+
+import { getMovieYear } from 'utils/helpers'
+import { getMovie, getSimilarResults } from './actions'
+import { selectMovie } from './selectors'
 
 import './Movie.scss';
 
 export class Movie extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      data: null
-    };
-  }
-
   componentDidMount() {
-    this.fetchMovie();
-  }
-
-  componentDidUpdate(prevProps) {
-    document.body.scrollTop = 0;
-    const previosMovieId = prevProps.match.params.movieId;
     const { movieId } = this.props.match.params;
-
-    if (previosMovieId !== movieId) {
-      this.fetchMovie();
-    }
+    this.props.getMovie(movieId)
   }
 
-  fetchMovie = () => {
-    const { movieId } = this.props.match.params;
-    fetch(`https://reactjs-cdp.herokuapp.com/movies/${movieId}`)
-      .then(response => response.json())
-      .then(data => {
-        this.setState({data: data});
-        window.scrollTo(0, 0);
-      });
+  componentDidUpdate() {
+    if (this.props.movie) {

```

Comment:
I believe we should check movie changes. Because component could be updated when props are not changed but some parent component changed. It will cause redundant requests.

Reply:
fixed, thank you!